Bottom: 1e6942b631d5e392b182fb4fa558d12b5a8a660f
Top:    f2fac2026f860d78c964260b4edd548ddd8d51c6
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-03-05 16:30:23 +0000

Refresh of attempt-to-refactor-the-step

---

diff --git a/src/hid/step/step_writer.c b/src/hid/step/step_writer.c
index 0a04ea0..79c492f 100644
--- a/src/hid/step/step_writer.c
+++ b/src/hid/step/step_writer.c
@@ -22,7 +22,10 @@ fprint_id_list (FILE *f, step_id_list list)
        iter = g_list_next (iter)) {
     fprintf (f, "#%i, ", GPOINTER_TO_INT (iter->data));
   }
-  fprintf (f, "#%i )", GPOINTER_TO_INT (iter->data));
+  if (iter == NULL)
+    fprintf (f, ")");
+  else
+    fprintf (f, "#%i )", GPOINTER_TO_INT (iter->data));
 }
 
 static void
